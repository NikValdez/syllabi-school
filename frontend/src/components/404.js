import React from 'react'

const NoMatch = ({ location }) => (
  <div>
    <h1>404</h1>
    <h2>
      Sorry! No page at address <code>{location.pathname}</code>
    </h2>
  </div>
)

export default NoMatch
