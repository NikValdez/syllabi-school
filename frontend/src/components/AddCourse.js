import gql from 'graphql-tag'
import React, { Component } from 'react'
import { Mutation } from 'react-apollo'
import styled from 'styled-components'
import Error from './ErrorMessage'
import { CURRENT_USER_QUERY_COURSES_EVENTS } from './MyCourses'

const AddButton = styled.button`
  justify-self: end;
  background: #f9c321;
  color: white;
  font-weight: 500;
  border: 0;
  border-radius: 0;
  text-transform: uppercase;
  font-size: 1rem;
  padding: 0.3rem 0.8rem;
  transform: skew(-2deg);
  transition: all 0.5s;
  &[disabled] {
    opacity: 0.5;
  }
`

const ADD_COURSE_MUTATION = gql`
  mutation ADD_COURSE_MUTATION($id: ID!) {
    addCourseToUser(id: $id) {
      id
    }
  }
`

class AddCourse extends Component {
  render() {
    const { id } = this.props

    return (
      <Mutation
        mutation={ADD_COURSE_MUTATION}
        variables={{
          id
        }}
        refetchQueries={[{ query: CURRENT_USER_QUERY_COURSES_EVENTS }]}
      >
        {(addCourseToUser, { loading, error }) => {
          if (error) return <Error error={error} />
          return (
            <AddButton disabled={loading} onClick={addCourseToUser}>
              Add{loading && 'ing'} âž•
            </AddButton>
          )
        }}
      </Mutation>
    )
  }
}

export default AddCourse
