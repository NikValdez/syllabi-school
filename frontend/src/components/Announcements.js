import gql from 'graphql-tag'
import htmlToText from 'html-to-text'
import moment from 'moment'
import React, { Component } from 'react'
import { Mutation, Query } from 'react-apollo'
import ReactModal from 'react-modal'
import styled, { keyframes } from 'styled-components'
import './styles/Modal.css'
import { TableStyles, TdStyles, ThStyles, TrStyles } from './styles/Table'

const stretch = keyframes`
  0% {
    transform: scale(.3);
    background-color: #7d5f05;
    border-radius: 100%;
  }

   100% {
    transform: scale(.5);
    background-color: #f9c321;
    border-radius: 100%;
  }
`
const Alert = styled.div`
  height: 50px;
  width: 50px;
  margin: 0 auto;
  background-color: #f9c321;
  animation-name: ${stretch};
  animation-duration: 1.5s;
  animation-timing-function: ease-out;
  animation-delay: 0;
  animation-direction: alternate;
  animation-iteration-count: infinite;
  animation-fill-mode: none;
  animation-play-state: running;
  font-size: 30px;
  box-shadow: 0 0 5px 3px rgba(0, 0, 0, 0.05);
`

const ANNOUNCEMENTS_QUERY = gql`
  query {
    me {
      id
      announcements {
        id
        date
        clicked
        text
      }
    }
  }
`

const UPDATE_ANNOUNCEMENT_MUTATION = gql`
  mutation UPDATE_ANNOUNCEMENT_MUTATION($id: ID!, $clicked: Boolean) {
    updateAnnouncement(id: $id, clicked: $clicked) {
      id
      clicked
    }
  }
`

class Announcements extends Component {
  state = {
    showModal: false
  }

  handleOpenAnnouncement = () => {
    this.setState({ showModal: true })
  }

  handleCloseModal = () => {
    this.setState({ showModal: false })
  }

  showDropdownMenu = e => {
    e.preventDefault()
    this.setState({ displayMenu: true }, () => {
      document.addEventListener('click', this.hideDropdownMenu)
    })
  }

  hideDropdownMenu = () => {
    this.setState({ displayMenu: false }, () => {
      document.removeEventListener('click', this.hideDropdownMenu)
    })
  }

  render() {
    return (
      <Query query={ANNOUNCEMENTS_QUERY}>
        {({ error, loading, data }) => {
          if (error) return <p>Error</p>
          if (loading) return <p>Loading...</p>
          if (!data.me.announcements) return <p>No Announcements</p>
          const announcements = data.me.announcements
            .sort((a, b) => {
              return new Date(a.date).getTime() - new Date(b.date).getTime()
            })
            .reverse()

          //get current count for number of announcements
          const count = announcements
            .map(announcement => {
              if (announcement.clicked === true) {
                return +1
              }
            })
            .filter(Boolean)

          return (
            <div>
              <div onClick={this.handleOpenAnnouncement}>
                {count.length > 0 && (
                  <Alert className="element">{count.length} </Alert>
                )}
                <span
                  style={{ position: 'absolute', top: '2rem', right: '2rem' }}
                >
                  üîî
                </span>
              </div>
              <ReactModal
                isOpen={this.state.showModal}
                contentLabel="modal"
                overlayClassName="Overlay"
                onRequestClose={this.handleCloseModal}
                shouldCloseOnOverlayClick={true}
                className="announcement-modal"
              >
                <>
                  {announcements.length < 1 ? (
                    <p style={{ background: '#f9c321' }}>
                      No Announcements Currently
                    </p>
                  ) : (
                    <TableStyles
                      style={{ border: '1px solid black', zIndex: '-10' }}
                    >
                      <tbody style={{ background: 'black' }}>
                        <tr>
                          <ThStyles style={{ color: 'white' }}>
                            Announcement
                          </ThStyles>
                          <ThStyles style={{ color: 'white' }}>Date</ThStyles>
                          <ThStyles style={{ color: 'white' }}>Seen</ThStyles>
                        </tr>

                        {announcements.map(({ text, date, id, clicked }) => (
                          <Mutation
                            mutation={UPDATE_ANNOUNCEMENT_MUTATION}
                            variables={{
                              id: id,
                              clicked: false
                            }}
                            key={id}
                          >
                            {(updateAnnouncement, { loading, error }) => (
                              <TrStyles key={id}>
                                <TdStyles style={{ color: 'white' }}>
                                  {htmlToText.fromString(text)}
                                </TdStyles>
                                <TdStyles style={{ color: 'white' }}>
                                  {moment(date).format('MMM Do YYYY')}
                                </TdStyles>
                                <TdStyles style={{ color: 'white' }}>
                                  <div
                                    onClick={() => {
                                      updateAnnouncement()
                                    }}
                                  >
                                    {clicked ? (
                                      <span style={{ marginLeft: '10px' }}>
                                        ‚ùå
                                      </span>
                                    ) : (
                                      <span style={{ marginLeft: '10px' }}>
                                        ‚òëÔ∏è
                                      </span>
                                    )}
                                  </div>
                                </TdStyles>
                              </TrStyles>
                            )}
                          </Mutation>
                        ))}
                      </tbody>
                    </TableStyles>
                  )}
                </>
              </ReactModal>
            </div>
          )
        }}
      </Query>
    )
  }
}

export default Announcements
export { ANNOUNCEMENTS_QUERY }
