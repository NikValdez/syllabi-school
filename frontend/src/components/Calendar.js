import moment from 'moment'
import React, { Component } from 'react'
import { Query } from 'react-apollo'
import BigCalendar from 'react-big-calendar'
import 'react-big-calendar/lib/css/react-big-calendar.css'
import { Modal } from 'react-bootstrap'
import ReactHtmlParser from 'react-html-parser'
import { CURRENT_USER_QUERY_COURSES_EVENTS } from './MyCourses'
import BigCalStyles from './styles/BigCalStyles'

const localizer = BigCalendar.momentLocalizer(moment)

class Calendar extends Component {
  state = {
    show: false,
    title: '',
    description: '',
    start: '',
    end: '',
    upload: ''
  }

  handleClose = () => {
    this.setState({ show: false })
  }

  handleShow = () => {
    this.setState({ show: true })
  }
  details = event => {
    this.setState({
      title: event.title,
      description: event.description,
      start: event.start,
      end: event.end,
      upload: event.upload
    })
  }
  render() {
    return (
      <>
        <Query query={CURRENT_USER_QUERY_COURSES_EVENTS}>
          {({ data, error, loading }) => {
            if (loading) return <p />
            if (error) return <p>Error : {error.message}</p>
            const courseData = data.me.myCourses.map(course => course.courses)
            const eventData = courseData.map(course => course.events)
            const calEvents = [].concat.apply([], eventData)

            return (
              <BigCalStyles>
                <BigCalendar
                  popup
                  onSelectEvent={event => {
                    this.handleShow()
                    this.details(event)
                  }}
                  views={['month']}
                  localizer={localizer}
                  events={calEvents}
                  startAccessor="start"
                  endAccessor="end"
                  eventPropGetter={events => ({
                    style: {
                      backgroundColor: events.color
                    }
                  })}
                />
              </BigCalStyles>
            )
          }}
        </Query>

        <Modal
          show={this.state.show}
          onHide={this.handleClose}
          style={{
            height: '40%',
            width: '40%',
            top: '50%',
            left: ' 50%',
            overflow: 'hidden'
          }}
        >
          <Modal.Header
            closeButton
            style={{
              width: '38%'
            }}
          >
            <Modal.Title>{this.state.title}</Modal.Title>
          </Modal.Header>
          <Modal.Body>
            <h4>Description:</h4> {ReactHtmlParser(this.state.description)}
            <h4>Upload:</h4>
            <p>
              <a href={this.state.upload} target="blank">
                {this.state.title} -- upload
              </a>
            </p>
            <h4>Dates:</h4>
            <p>
              {moment(this.state.start).format('MMM Do YYYY')} -{' '}
              {moment(this.state.end).format('MMM Do YYYY')}
            </p>
          </Modal.Body>
        </Modal>
      </>
    )
  }
}

export default Calendar
