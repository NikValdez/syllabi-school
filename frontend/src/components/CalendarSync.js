/* global gapi */
import htmlToText from 'html-to-text'
import ical from 'ical-generator'
import React, { Component } from 'react'
import { Modal } from 'react-bootstrap'
import Button from './styles/Button'

gapi.load('client:auth2', function() {
  gapi.auth2.init({
    client_id:
      '335751011458-j6070j2ugp7a915kv6t6u6s848s0bp12.apps.googleusercontent.com'
  })
})

export default class CalendarSync extends Component {
  state = {
    courseId: this.props.courseId,
    appleEvents: this.props.courseEvents,
    googleEvents: this.props.courseEvents,
    show: false,
    linkUrl: ''
  }

  handleClose = () => {
    this.setState({ show: false })
  }

  handleShow = () => {
    this.setState({ show: true })
  }

  createIcal = () => {
    this.state.appleEvents.map(course => {
      course.summary = course.title
      delete course.title
      course.description = htmlToText.fromString(course.description)
      course.background = course.color
      delete course.color

      this.setState({
        appleEvents: this.state.appleEvents
      })
    })
    const calendar = ical({
      domain: 'gosyllabi.com',
      prodId: '//Syllabi//ical-generator//EN',
      events: this.props.courseEvents
    }).toURL()

    let link = document.createElement('a')
    document.body.appendChild(link)
    link.download = `${this.props.courseTitle}- schedule`
    link.href = calendar
    return link.click()
  }

  createOulookCal = () => {
    this.state.appleEvents.map(course => {
      course.summary = course.title
      delete course.title
      course.description = htmlToText.fromString(course.description)
      course.background = course.color
      delete course.color

      this.setState({
        appleEvents: this.state.appleEvents
      })
    })
    const calendar = ical({
      domain: 'gosyllabi.com',
      prodId: '//Syllabi//ical-generator//EN',
      events: this.props.courseEvents
    })

    return window.open('data:text/calendar;charset=utf8,' + escape(calendar))
  }

  authenticate = () => {
    return gapi.auth2
      .getAuthInstance()
      .signIn({
        scope:
          'https://www.googleapis.com/auth/calendar https://www.googleapis.com/auth/calendar.events'
      })
      .then(
        function() {
          console.log('Sign-in successful')
        },
        function(err) {
          console.error('Error signing in', err)
        }
      )
      .then(this.loadClient)
  }

  loadClient = () => {
    return gapi.client
      .load('https://content.googleapis.com/discovery/v1/apis/calendar/v3/rest')
      .then(
        function() {
          console.log('GAPI client loaded for API')
        },
        function(err) {
          console.error('Error loading GAPI client for API', err)
        }
      )
      .then(this.execute)
  }

  execute = () => {
    this.state.googleEvents.map(course => {
      course.summary = course.title
      delete course.title
      course.background = course.color
      delete course.color
      course.start = { dateTime: course.start }
      course.end = { dateTime: course.end }
      delete course.id
      this.setState({
        googleEvents: this.state.googleEvents
      })
    })
    this.state.googleEvents.map(event => {
      return gapi.client.calendar.events
        .insert({
          calendarId: 'primary',
          resource: event
        })
        .then(
          response => {
            this.setState({
              linkUrl: response.result.htmlLink,
              show: true
            })
          },
          err => {
            console.error('Execute error', err)
          }
        )
    })
  }

  addScheduleToCalendar = () => {
    window.open(this.state.linkUrl)
  }

  render() {
    return (
      <div>
        <h4>Sync:</h4>
        <h4
          onClick={this.createIcal}
          style={{
            display: 'inline-block',
            cursor: 'pointer',
            marginBottom: '25px'
          }}
        >
          <i className="far fa-calendar-alt" /> Apple
        </h4>
        <h4
          onClick={this.authenticate}
          style={{
            display: 'inline-block',
            marginLeft: '10px',
            marginBottom: '15px',
            cursor: 'pointer'
          }}
        >
          <i className="far fa-calendar-alt" /> Google
        </h4>
        {/* <h4
          onClick={this.createOulookCal}
          style={{
            display: 'inline-block',
            cursor: 'pointer',
            marginLeft: '10px'
          }}
        >
          <i className="far fa-calendar-alt" /> Outlook
        </h4> */}

        <Modal
          show={this.state.show}
          onHide={this.handleClose}
          style={{
            height: '30%',
            width: '40%',
            top: '50%',
            left: ' 50%',
            overflow: 'hidden',
            minHeight: 0
          }}
        >
          <Modal.Header
            closeButton
            style={{
              width: '38%'
            }}
          >
            <h2>Add Schedule to Google Calendar</h2>
          </Modal.Header>
          <Modal.Body>
            <Button
              onClick={() => {
                this.addScheduleToCalendar()
                this.handleClose()
              }}
            >
              Add to <i className="fab fa-google" />
              oogle
            </Button>
          </Modal.Body>
        </Modal>
      </div>
    )
  }
}
