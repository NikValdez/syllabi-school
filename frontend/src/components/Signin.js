import gql from 'graphql-tag'
import React, { Component } from 'react'
import { Mutation } from 'react-apollo'
import { Link } from 'react-router-dom'
import Error from './ErrorMessage'
import { CURRENT_USER_QUERY } from './User'
const SIGNIN_MUTATION = gql`
  mutation SIGNIN_MUTATION($email: String!, $password: String!) {
    signin(email: $email, password: $password) {
      id
      email
      name
    }
  }
`

class Signin extends Component {
  state = {
    email: '',
    password: ''
  }
  saveToState = e => {
    this.setState({ [e.target.name]: e.target.value })
  }
  render() {
    return (
      <Mutation
        mutation={SIGNIN_MUTATION}
        variables={this.state}
        refetchQueries={[{ query: CURRENT_USER_QUERY }]}
      >
        {(signin, { error, loading }) => (
          <>
            <form
              method="post"
              onSubmit={async e => {
                e.preventDefault()
                await signin()
                this.setState({ email: '', password: '' })
              }}
            >
              <fieldset disabled={loading} aria-busy={loading}>
                <h1 class="title is-spaced">Account</h1>
                <p className="has-text-right">
                  <Link to="/signup" className="">
                    Need an account? Sign Up
                  </Link>
                </p>

                <Error error={error} />

                <div className="field">
                  <label className="label" htmlFor="email">
                    Email
                    <input
                      className="input"
                      type="email"
                      name="email"
                      placeholder="email"
                      value={this.state.email}
                      onChange={this.saveToState}
                    />
                  </label>
                </div>

                <div className="field">
                  <label className="label" htmlFor="password">
                    Password
                    <input
                      className="input"
                      type="password"
                      name="password"
                      placeholder="password"
                      value={this.state.password}
                      onChange={this.saveToState}
                    />
                  </label>

                  <p className="has-text-right">
                    <Link to="/request_reset">Forgot Password?</Link>
                  </p>
                </div>

                <div className="field">
                  <button
                    className="button is-fullwidth is-medium is-black"
                    type="submit"
                  >
                    Sign In
                  </button>
                </div>
              </fieldset>
            </form>
          </>
        )}
      </Mutation>
    )
  }
}

export default Signin
export { SIGNIN_MUTATION }
