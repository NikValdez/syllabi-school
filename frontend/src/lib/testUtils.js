import casual from 'casual'

casual.seed(777)
const fakeUser = () => ({
  __typename: 'User',
  id: '4234',
  name: casual.name,
  email: casual.email,
  permissions: ['ADMIN'],
  institution: casual.title
})

export { fakeUser }
