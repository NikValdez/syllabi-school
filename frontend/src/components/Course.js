import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import DeleteCourse from './DeleteCourse'
import AddCourse from './AddCourse'

class Course extends Component {
  render() {
    const { course } = this.props
    console.log(course.id)
    return (
      <div>
        <Link to={`/courses/${course.id}`}>
          <h3>{course.title}</h3>
        </Link>
        <h6>{course.courseCode}</h6>
        <p>{course.description}</p>
        <p>Credits: {course.credits}</p>
        <DeleteCourse id={course.id}>Delete ‚ùå</DeleteCourse>
        <AddCourse id={course.id} />
      </div>
    )
  }
}

export default Course
