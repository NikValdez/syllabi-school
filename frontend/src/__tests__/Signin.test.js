import { mount } from 'enzyme'
import toJSON from 'enzyme-to-json'
import React from 'react'
import { ApolloConsumer } from 'react-apollo'
import { MockedProvider } from 'react-apollo/test-utils'
import { BrowserRouter as Router } from 'react-router-dom'
import wait from 'waait'
import Signin, { SIGNIN_MUTATION } from '../components/Signin'
import { CURRENT_USER_QUERY } from '../components/User'
import { fakeUser } from '../lib/testUtils'

function type(wrapper, name, value) {
  wrapper.find(`input[name="${name}"]`).simulate('change', {
    target: { name, value }
  })
}

const me = fakeUser()
const mocks = [
  // signin mock mutation
  {
    request: {
      query: SIGNIN_MUTATION,
      variables: {
        email: me.email,
        password: 'password'
      }
    },
    result: {
      data: {
        signin: {
          __typename: 'User',
          id: 'abc123',
          email: me.email
        }
      }
    }
  },
  // current user query mock
  {
    request: { query: CURRENT_USER_QUERY },
    result: { data: { me } }
  }
]

describe('<Signin/>', () => {
  it('renders and matches snapshot', async () => {
    const wrapper = mount(
      <Router>
        <MockedProvider>
          <Signin />
        </MockedProvider>
      </Router>
    )
    expect(toJSON(wrapper.find('form'))).toMatchSnapshot()
    expect(wrapper.find('form').length).toBe(1)
    expect(wrapper.find('Link').length).toBe(2)
  })

  it('calls the mutation properly', async () => {
    let apolloClient
    const wrapper = mount(
      <Router>
        <MockedProvider mocks={mocks}>
          <ApolloConsumer>
            {client => {
              apolloClient = client
              return <Signin onSubmit={jest.fn()} />
            }}
          </ApolloConsumer>
        </MockedProvider>
      </Router>
    )
    await wait()
    wrapper.update()
    // console.log(wrapper.debug())
    type(wrapper, 'email', me.email)
    type(wrapper, 'password', 'password')
    wrapper.update()
    wrapper.setState({ email: me.email, password: 'password' })
    expect(wrapper.state('email')).toEqual(me.email)
    expect(wrapper.state('password')).toEqual('password')

    await wait()
    wrapper.update()
    wrapper.find('form').simulate('submit')
    await wait()
    wrapper.update()
    // expect(jest.fn()).toHaveBeenCalledTimes(1)

    wrapper.update()

    // // query the user out of the apollo client
    // const user = await apolloClient.query({ query: CURRENT_USER_QUERY })
    // expect(user.data.me).toMatchObject(me)
  })
})

// expect(testValues.handleSubmit).toBeCalledWith({username: testValues.username, password: testValues.password});
