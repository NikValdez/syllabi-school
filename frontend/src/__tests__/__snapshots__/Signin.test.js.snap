// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signin/> renders and matches snapshot 1`] = `
<form
  className="login-form sc-bwzfXH ipuXRf"
  method="post"
  onSubmit={[Function]}
>
  <fieldset
    aria-busy={false}
    disabled={false}
  >
    <h2>
      Sign In
    </h2>
    <ErrorMessage
      error={Object {}}
    />
    <label
      htmlFor="email"
    >
      Email
      <input
        name="email"
        onChange={[Function]}
        placeholder="email"
        type="email"
        value=""
      />
    </label>
    <label
      htmlFor="password"
    >
      Password
      <input
        name="password"
        onChange={[Function]}
        placeholder="password"
        type="password"
        value=""
      />
    </label>
    <button
      type="submit"
    >
      Sign In!
    </button>
  </fieldset>
  <Link
    replace={false}
    style={
      Object {
        "textDecoration": "none",
      }
    }
    to="/signup"
  >
    <a
      href="/signup"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <button
        style={
          Object {
            "margin": "0 auto",
            "marginTop": "10px",
            "width": "80%",
          }
        }
      >
        Sign Up
      </button>
    </a>
  </Link>
  <Link
    replace={false}
    style={
      Object {
        "textDecoration": "none",
      }
    }
    to="/request_reset"
  >
    <a
      href="/request_reset"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <button
        style={
          Object {
            "margin": "0 auto",
            "marginTop": "10px",
            "textDecoration": "none",
            "width": "70%",
          }
        }
      >
        Forgot Password?
      </button>
    </a>
  </Link>
</form>
`;
